### PR TITLE
Fix upgrade script and make parity run when no .parity dir.

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

**Comment by General-Beck** [comment]
check first run on clean install
`ERROR: Error upgrading parity data: CannotLockVersionFile`
workaround: run `mkdir -p ~/.parity` before running parity

File: parity/upgrade.rs [hunk 1]
```diff
@@ -18,14 +18,15 @@
...
-use std::fs::File;
+use std::fs::{File, create_dir_all};
 use std::env;
 use std::io::{Read, Write};
 
 #[cfg_attr(feature="dev", allow(enum_variant_names))]
 #[derive(Debug)]
 pub enum Error {
-	CannotLockVersionFile,
+	CannotCreateConfigPath,
```
Comment (tomusdrw) [inline parity/upgrade.rs]: Wrapping std::io::Error in those errors could help debugging potential problems.

**Comment by General-Beck** [comment]
First run
`Apr 14 22:48:45 localhost ubuntu-core-launcher[2912]: ERROR: Error upgrading parity data: CannotCreateConfigPath`

`Apr 14 22:48:45 localhost kernel: [  720.307940] audit: type=1400 audit(1460674125.038:72): apparmor="DENIED" operation="mkdir" profile="parity.sideload_parity_IRKcLLeOIDVJ" name="/root/.parity/" pid=2912 comm="parity" requested_mask="c" denied_mask="c" fsuid=0 ouid=0`
 
#set data dir`
PARITY_HOME=${SNAP_APP_DATA_PATH}
export PARITY_HOME
 #check first run - create data dir
$verbose && echo "parity home:" $PARITY_HOME
mkdir -p $PARITY_HOME/.parity
 # start parity daemon
exec ${SNAP_APP_PATH}/bin/$plat_abi/parity -d "$PARITY_HOME/.parity" 

Need to add a condition that checks to see if the --db-path

**Comment by gavofyork** [comment]
actually, it's not clear what the path should be for the config file.

**Comment by NikVolf** [comment]
should also add `-keys-path $PARITY_HOME/.parity/keys`

**Comment by General-Beck** [comment]
and --password $PARITY_HOME/.parity/pass

**Pull request merged**